addons/account/static/src/js/reconciliation/reconciliation_model.js

```diff
@@ -1388,9 +1388,8 @@ var ManualModel = StatementModel.extend({
                 default:
                     var partner_ids = context.partner_ids;
                     var account_ids = context.account_ids || self.account_ids;
-                    if (partner_ids && !account_ids) account_ids = [];
-                    if (!partner_ids && account_ids) partner_ids = [];
-                    account_ids = null; // TOFIX: REMOVE ME
+                    if (partner_ids && !account_ids) account_ids = null;
+                    if (!partner_ids && account_ids) partner_ids = null;
```

Comment:
according to the docstring of method 'get_all_data_for_manual_reconciliation'
""" Returns the data required for the invoices & payments matching of partners/accounts.
            If an argument is None, fetch all related reconciliations. Use [] to fetch nothing.
        """
Hence by looking at what the code was doing previously, we can assume that it was always fetching the reconciliation for the accounts and it would have fetched the reconciliation for the partner only if partner_ids was defined in the context. I believe this is wrong as it is not consistent with the fact that in this case we want to show reconciliation for all partners and accounts.

I think that the fix could be rewritten like this to take into account all use case:

```
default:
    var partner_ids = context.partner_ids || null;
    var account_ids = context.account_ids || self.account_ids || null;
    return self._rpc({
    ...
```
If the context contains an empty list, it would stay that way, but if nothing is defined in the context, the value would be undefined and undefined || null will result to null;

Reply:
Yes, I think you are right. I also amended the test file, not sure about that...

Reply:
seems correct, just gonna wait to see if the test are green :)